http-common/src/connector.rs

```diff
@@ -461,25 +402,102 @@ impl std::error::Error for ConnectorError {
 	}
 }
 
+/// Finds the systemd socket if one has been used to socket-activate this process.
+///
+/// This mimics `sd_listen_fds` from libsystemd, then returns the very first fd.
+fn get_systemd_socket() -> Result<Option<std::os::unix::io::RawFd>, String> {
+	// Ref: <https://www.freedesktop.org/software/systemd/man/sd_listen_fds.html>
+	//
+	// >sd_listen_fds parses the number passed in the $LISTEN_FDS environment variable, then sets the FD_CLOEXEC flag
+	// >for the parsed number of file descriptors starting from SD_LISTEN_FDS_START. Finally, it returns the parsed number.
+	//
+	// Note that this function always returns the first fd. It cannot be used for processes which expect more than one socket.
+	// CS/IS/KS only expect one socket, so this is fine, but it is not the case for iotedged (mgmt and workload sockets) for example.
+	//
+	// If obtaining more than one fd is required in the future, keep in mind that it requires getting fds by name (by inspecting the LISTEN_FDNAMES env var)
+	// instead of by number, since systemd does not pass down multiple fds in a deterministic order. The complication with LISTEN_FDNAMES is that
+	// CentOS 7's systemd is too old and doesn't support it, which would mean CS/IS/KS would have to forego systemd socket activation on CentOS 7
```

Comment:
Very nit (I'm sorry): The correct word here should be "forgo". "forgo" = "to do without"; "forego" = "to precede"

Although I just googled it and it seems like enough people mix them up that "forego" is now defined as a variant spelling of "forgo".

Reply:
TIL. Fixed.

Reply:
TIL too! 😄